k8s-deploy-kubevirt.sh

```diff
@@ -3,8 +3,7 @@
 set -ex
 
 # Install CDI
-LATEST_CDI=$(curl -s https://api.github.com/repos/kubevirt/containerized-data-importer/releases/latest)
-export CDI_VERSION=$(jq -r .tag_name <<< $LATEST_CDI)
+[ ! -v CDI_VERSION ] && export CDI_VERSION=v1.55.2
```

Comment:
how about checking if there is a given env var and in case it's not there try to use curl(from github actions)?

e.g:
```
[ ! -v $CDI_VERSION] && LATEST_CDI=$(curl -s https://api.github.com/repos/kubevirt/containerized-data-importer/releases/latest) && export CDI_VERSION=$(jq -r .tag_name <<< $LATEST_CDI)
```

so if we run it locally, we do try to get it using curl.

Reply:
@liranr23 I see two issues with this approach:
1. we go back to unauthenticated requests through github api which have lower api rate limit
2. we already have this issue that we don't pin the versions of our dependencies, either bazel or cdi. we don't want out PRs to fail when bazel or CDI break backward compatibility - these tests should verify the changes we do. when we have integration tests, then it would make more sense to take the latest version of dependencies that should  be supported, but as we still don't have integration tests - I'm ok with keep using the latest version of CDI in those tests. as for local execution - I think it's also a flow that predictability and reproducibility is more important than catch some failure with latest version of a dependency

Reply:
> @liranr23 I see two issues with this approach:
> 
> 1. we go back to unauthenticated requests through github api which have lower api rate limit

Well, if it's not provided from the github runner, then yes. But if we run locally I think it's fine overall.

> 2. we already have this issue that we don't always pin the versions of our dependencies, either bazel (that was fixed) or cdi. we don't want our PRs to fail when bazel or CDI break backward compatibility - these tests should verify the changes we do. when we have integration tests, then it would make more sense to take the latest version of dependencies that should  be supported, but as we still don't have integration tests - I'm ok with keep using the latest version of CDI in those tests. as for local execution - I think it's also a flow that predictability and reproducibility is more important than catching some failure with latest version of a dependency

OK, this is a good argument :)